src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -34,33 +34,73 @@ end
 # 
 
 """
-    svdTTN(os::OpSum{C}, sites::IndsNetwork{V<:Index}, root_vertex::V, kwargs...) where {C,V}
+    ttn_svd(os::OpSum, sites::IndsNetwork, root_vertex, kwargs...)
 
 Construct a dense TreeTensorNetwork from a symbolic OpSum representation of a
 Hamiltonian, compressing shared interaction channels.
 """
-function svdTTN(
-  os::OpSum{C}, sites::IndsNetwork{VT,<:Index}, root_vertex::VT; kwargs...
-)::TTN where {C,VT}
-  mindim::Int = get(kwargs, :mindim, 1)
-  maxdim::Int = get(kwargs, :maxdim, 10000)
-  cutoff::Float64 = get(kwargs, :cutoff, 1e-15)
+function ttn_svd(
+  os::OpSum, sites::IndsNetwork, root_vertex; kwargs...
+)
+  # Function barrier to improve type stability
+  coefficient_type = ITensors.determineValType(terms(os))
+  return ttn_svd(coefficient_type, os, sites, root_vertex; kwargs...)
+end
 
-  ValType = ITensors.determineValType(ITensors.terms(os))
+function ttn_svd(
+  coefficient_type::Type{<:Number},
+  os::OpSum,
+  sites::IndsNetwork,
+  root_vertex;
+  mindim::Int=1,
+  maxdim::Int=typemax(Int),
+  cutoff=eps(coefficient_type) * 10,
```

Comment:
Still think this should be `eps(real(coefficient_type))` to handle complex `coefficient_type`.